File: Abstraction/Entities/Product.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Abstraction.Entities;

[Table("Product")]
public class Product
    : BaseEntity
{
    public Product()
        : base()
    {
    }

    public Product(int id)
        : base(id)
    {
    }

    [Column("ProductCategoryId")]
    [ForeignKey(nameof(Category))]
    public int ProductCategoryId { get; set; }

    [Column("ProductName")]
    public string ProductName { get; set; }

    [Column("Price")]
    public decimal Price { get; set; }

    public virtual ProductCategory Category { get; set; }

    public virtual ICollection<ReceiptDetail> ReceiptDetails { get; init; }
}

File: angular/src/app/shared/models/product.ts
import { BaseModel } from './base.model';

export class Product extends BaseModel {
  productCategoryId: number;
  categoryName: string;
  productName: string;
  price: number;
  receiptDetailIds: number[];

  constructor(
    id: number = 0,
    productCategoryId: number = 0,
    categoryName: string = '',
    productName: string = '',
    price: number = 0,
    receiptDetailIds: number[] = []
  ) {
    super(id);
    this.productCategoryId = productCategoryId;
    this.categoryName = categoryName;
    this.productName = productName;
    this.price = price;
    this.receiptDetailIds = receiptDetailIds;
  }
}
